core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1560,26 +1567,30 @@ private[spark] class BlockManager(
   def replicateBlock(
       blockId: BlockId,
       existingReplicas: Set[BlockManagerId],
-      maxReplicas: Int): Unit = {
+      maxReplicas: Int,
+      forceFetch: Boolean = true,
+      maxReplicationFailures: Option[Int] = None): Boolean = {
+    var replicatedSuccessfully = true
```

Comment:
Personally, I don't like var's we could factor this out by using map + headOption + getOrElse but that might not be any cleaner so no pressure on this suggestions.

Reply:
fixed.